pandas/core/generic.py

```diff
@@ -2015,8 +2015,17 @@ def __array__(
         self, dtype: npt.DTypeLike | None = None, copy: bool | None = None
     ) -> np.ndarray:
         values = self._values
-        arr = np.asarray(values, dtype=dtype)
-        if astype_is_view(values.dtype, arr.dtype) and self._mgr.is_single_block:
+        if copy is None:
+            # Note: branch avoids `copy=None` for NumPy 1.x support
+            arr = np.asarray(values, dtype=dtype)
+        else:
+            arr = np.array(values, dtype=dtype, copy=copy)
+
+        if (
+            copy is not False
```

Comment:
and also not `True`, because then it will already have copied in `arr = np.array(values, dtype=dtype, copy=copy)` above?

Reply:
We pass on `copy=copy`, so we can trust the semantics being correct until here.

Reply:
Hmmmm, but I think I got it wrong...  It should rather say, `copy is not True`?  Because if we force a copy it doesn't seem sensible to create a read-only view.

(Have to see if I can write a test for these corrections, it is clearly trickier then just err'ing on the "raise an error" side!)

Reply:
Yes, that is what I meant